wagtail/core/hooks.py

```diff
@@ -32,6 +33,48 @@ def decorator(fn):
     _hooks[hook_name].append((fn, order))
 
 
+class TemporaryHook(ContextDecorator):
```

Comment:
I like the idea, but should this not be in `tests/utils/hooks.py`, similar to `tests/utils/form_data.py`?

Unless there's wider scope to support this in core..

Reply:
I think it's best to keep all registration logic in the main hooks module. It directly modifies the `_hooks` global variable that I think should be considered internal to the hooks module.

Also, I plan to document this, third party modules could use it for testing their custom hooks.